### PR TITLE
Isconnecting

### PR DESCRIPTION
**Pull request by tapmantwo**
…ing to establish a connection.

This is to assist with issue reported: #783

I'm not sure on best approach to write unit tests for this particular issue - if you have pointers, let me know and I'll happily add them in.

(This was previously submitted #786, but now updated to fix line ending probs).

### CODE REVIEW

File: StackExchange.Redis/StackExchange/Redis/PhysicalBridge.cs [hunk 1]
```diff
@@ -70,6 +70,8 @@ public enum State : byte
 
         public bool IsConnected => state == (int)State.ConnectedEstablished;
 
+        public bool IsConnecting => state == (int)State.ConnectedEstablished || state == (int)State.Connecting;
```
Comment (NickCraver) [inline StackExchange.Redis/StackExchange/Redis/PhysicalBridge.cs]: This one doesn't make sense to me, why is `state == (int)State.ConnectedEstablished` considered "connecting"? It's already the case for IsConnected above, I'd expect this to only return `state == (int)State.Connecting`

Comment (tapmantwo) [reply StackExchange.Redis/StackExchange/Redis/PhysicalBridge.cs]: The rationale was that if it's not yet `Connected`, it is still `Connecting`.  However if it's easier to understand I'd be happy to adjust, as long as there is a mechanism to determine that the multiplexor is currently in the process of connecting.

Comment (NickCraver) [reply StackExchange.Redis/StackExchange/Redis/PhysicalBridge.cs]: @tapmantwo Not sure I follow, are you saying `state == (int)State.ConnectedEstablished` is "not yet connected"? That's the part I'm having a disconnect on, as we read that as "IsConnected" above

Comment (tapmantwo) [reply StackExchange.Redis/StackExchange/Redis/PhysicalBridge.cs]: Oh my, that is a good spot!  It was supposed to be State.ConnectedEstablishing.  I will update now :)

**Pull request merged**